### ISSUE TITLE
Input data shuffling in multi-class classification problem

### ISSUE THREAD

**Issue by dlihhats** [JohnLangford/vowpal_wabbit#1160]
I'm observing a difference in the way VW works when launched from Bash and through Python sklearn.

I've got a multiclass classification problem with 550 classes. I have the train data in VW format with all samples sorted by label. When I pass this train set to VW in Bash with --oaa option, it's obvious that VW reads the data as a truly online learning algorithm, i.e. sequentially. As a result, it overfits and gives almost zero accuracy on the validation set.

At the same time, when I run VW through sklearn, it does the shuffling and the accuracy is just fine.
From sklearn_vw.py:
```
 # add examples to model
        for n in range(self.passes_):
            if n > 1:
                np.random.shuffle(X)
            for idx, x in enumerate(X):
                model.learn(x)
        self.fit_ = True
        return self
```

I've gone through the docs, and I can't find an option for VW to ask it to shuffle incoming file.

**Comment by JohnLangford** [comment]
There is no such option.  Typically, I use a commandline tool like 'shuf' or 'sort -R' if I want to shuffle on the commandline.
